src/com/amazon/corretto/crypto/provider/AesKeyWrapPaddingSpi.java

```diff
@@ -0,0 +1,304 @@
+    // appended.
+    private static int getWrappedLen(final int unwrappedLen) {
+        final int paddingLen;
+        if (unwrappedLen % 8 == 0) {
+            paddingLen = 0;
+        } else {
+            paddingLen = 8 - (unwrappedLen % 8);
+        }
+        return Math.addExact(Math.addExact(unwrappedLen, paddingLen), 8);
+    }
+
+    // Because the key is padded before it is wrapped (i.e. encrypted), we have
+    // no way of knowing the unwrapped key's precise size beforehand. The best
+    // we can do is to "guess" by accounting for the 8 bytes of padding that
+    // are added in all cases before wrapping.
+    private static int estimateUnwrappedLen(final int wrappedLen) {
+        if (wrappedLen < 16) {
+             return 8;
+        }
+        return Math.subtractExact(wrappedLen, 8);
+    }
+
+    @Override
+    protected byte[] engineGetIV() {
+        return null;
+    }
+
+    @Override
+    protected AlgorithmParameters engineGetParameters() {
+        return null;
+    }
+
+    @Override
+    protected void engineInit(int opmode, Key key, SecureRandom ignored)
+        throws InvalidKeyException {
+        implInit(opmode, key);
+    }
+
+    @Override
+    protected void engineInit(int opmode, Key key, AlgorithmParameters ignored1, SecureRandom ignored2)
+            throws InvalidKeyException {
+        implInit(opmode, key);
+    }
+
+    @Override
+    protected void engineInit(int opmode, Key key, AlgorithmParameterSpec ignored1, SecureRandom ignored2)
+            throws InvalidKeyException {
+        implInit(opmode, key);
+    }
+
+    private void implInit(int opmode, Key key) throws InvalidKeyException {
+        if (opmode != Cipher.UNWRAP_MODE && opmode != Cipher.WRAP_MODE
+                && opmode != Cipher.ENCRYPT_MODE && opmode != Cipher.DECRYPT_MODE) {
+            throw new UnsupportedOperationException("Unsupported mode");
+        }
+        if (key == null) {
+            throw new InvalidKeyException("Null key");
+        }
+        if (key != jceKey) {
+            if (!(key instanceof  SecretKey)) {
+                throw new InvalidKeyException("Need a SecretKey");
+            }
+            if (!"RAW".equalsIgnoreCase(key.getFormat())) {
+                throw new InvalidKeyException("Need a raw format key");
+            }
+            if (!"AES".equalsIgnoreCase(key.getAlgorithm())) {
+                throw new InvalidKeyException("Expected an AES key");
+            }
+            if (keyBytes != null) {
+                Arrays.fill(keyBytes, (byte) 0);
+                keyBytes = null;
+            }
+            keyBytes = key.getEncoded();
+            if (keyBytes == null) {
+                throw new InvalidKeyException("Key doesn't support encoding");
+            }
+            jceKey = (SecretKey) key;
+        }
+        this.opmode = opmode;
+        buffer.reset();
+    }
+
+    @Override
+    protected byte[] engineUpdate(byte[] in, int inOfs, int inLen) {
```

Comment:
Naming nit, I'd prefer the more readable `inOffset` rather than `inOfs` throughout this file.

Reply:
sure, i'll adjust that for `inOfs` and `outOfs`.